packages/cloud-functions/src/cico/rampWebhook.ts

```diff
@@ -0,0 +1,50 @@
+import crypto from 'crypto'
+import stableStringify from 'fast-json-stable-stringify'
+import * as functions from 'firebase-functions'
+import { readFileSync } from 'fs'
+import { RAMP_KEY } from '../config'
+import { saveTxHashProvider } from '../firebase'
+import { Provider } from './Provider'
+
+const rampKey = readFileSync(`./config/${RAMP_KEY}`).toString()
+
+function verifyRampSignature(signature: string | undefined, body: any) {
+  if (!signature || !body) {
+    return false
+  }
+
+  const verifier = crypto.createVerify('sha256')
+  verifier.update(Buffer.from(stableStringify(body)))
+  return verifier.verify(rampKey, signature, 'base64')
+}
+
+const RAMP_SIGNATURE_HEADER = 'X-Body-Signature'
+const RAMP_RELEASED = 'RELEASED'
+const RAMP_RELEASING = 'RELEASING'
```

Comment:
This is subjective so feel free to ignore but I don't find these variables very helpful given they are included in a file dedicated to Ramp, aren't exported, and aren't significantly more descriptive than the values themselves

Reply:
Using an enum for the request body type now :)